README.md

```diff
@@ -479,6 +509,89 @@ registry.define('my-widget', () => {
 });
 ```
 
+#### Injecting State
+
+Working with larger widget structures, it can quickly become tiresome and complex to pass all the required properties down the tree. Additionally having to do so often means that widgets are forced to be aware of properties that they they don't need, but are required to know about in order to ensure they are propagated to the children.
+
+Dojo 2 provides a mechanism to inject state directly to parts of the widget structure; this is done by defining an `Injector` in the `registry` and passing a context object that will source the state that is to be injected.
+
+```ts
+import { Injector, BaseInjector } from '@dojo/widget-core/Injector';
+
+const myStateContext = {
+	theme: 'solid'
+};
+
+registry.define('state', Injector(BaseInjector, myStateContext));
+```
+
+To use the injected state, create a `beforeRender` method that returns a render function which creates a `w` reference to the registry item like any other widget and pass the properties required by the `InjectorProperties` interface.
+
+```ts
+beforeRender(renderFunc: () => DNode, properties: any, children: any): DNode {
+	return () => {
+		return w('state', {
+			render: renderFunc,
+			getProperties(context: any, properties: any): any {
+				return context;
+			},
+			properties,
+			getChildren(context: any, children: DNode[]): DNode[] {
+				return [];
+			},
+			children
+		});
+	};
+}
+```
+
+This will inject the values of `myState` as properties to the widget, as the returned object from `getProperties` is mixed over the widgets' existing properties. 
+
+For convenience, Dojo 2 provides mixin called `Container` that will decorate a widget with the above `beforeRender` implementation. Using the `Container` mixin enables any view widget to have state injected without coupling the widget to always have state injected. This means the widget can also be used as a normal widget with properties being passed from it's parent.
+
+
+```ts
+import { MyViewWidget } from './MyViewWidget';
+import { Container } from '@dojo/widget-core/Container';
+
+function getProperties(context: any, properties: any): any {
+	return context;
+}
+
+const MyViewWidgetContainer = Container(MyViewWidget, 'state', { getProperties });
+```
+
+**Note:** that both the `getProperties` and `getChildren` functions do not need to be provided, if the functions are not defined the default mappers will be used that return an empty object and an empty array respectively.
```

Comment:
Should we have a section 'Default container implementation' that doesn't have the `getProperties` function, and then a section on 'Advanced container implementation' that is like the code below?